### PR TITLE
Port CUE SDK to Dagger Engine 0.3

### PR DESCRIPTION
**Pull request by jlongtine**
CUE SDK port

Starts the implementation of https://github.com/dagger/dagger/issues/3121

### CODE REVIEW

**Comment by shykes** [comment]
Awesome! Is this something we should plan to release, or the first in a series of PRs to get us to a releaseable state?

**Comment by jlongtine** [comment]
> Awesome! Is this something we should plan to release, or the first in a series of PRs to get us to a releaseable state?

That's an open question for me, as well, @shykes. I think that depends on a few factors.

Some context may be helpful?

This PR gets us to ~85%-90% coverage of the core Europa APIs. You can get more specifics on what is working/missing in https://github.com/dagger/dagger/issues/3121. 

TLDR, though, is: 
- Secrets support is not good (currently saving new Secrets as files... which has the plaintext in the ID + saves them in the buildkit cache -- I personally _would not_ release until we get this sorted)
- `core.#Push`, `core.#Pull`, and `core.GitPull` all are missing auth, which makes it rather difficult to use with private registries. We've seen users hitting this in the Go SDK, as well.
- Sockets are unsupported
- Cache declarations that used to work won't for now

All in all, it's basically as implemented as it can be, given the current state of the v0.3 engine. But, I'm confident there are a bunch of plans that will break if we release as is.

**My recommendation** would be the latter. 

I don't think we can consider this a replacement for Europa until things like `core.#Exec & {always: true}` work.

That said, I don't know when the things I'll need to get to a higher coverage or greater security will be available or prioritized.

File: solver/solver.go [hunk 1]
```diff
@@ -1,309 +1,33 @@
...
-func (s *Solver) ResolveImageConfig(ctx context.Context, ref string, opts llb.ResolveImageConfigOpt) (dockerfile2llb.Image, digest.Digest, error) {
-	var image dockerfile2llb.Image
-
-	// Load image metadata and convert to to LLB.
-	// Inspired by https://github.com/moby/buildkit/blob/master/frontend/dockerfile/dockerfile2llb/convert.go
-	// FIXME: this needs to handle platform
-	dg, meta, err := s.opts.Gateway.ResolveImageConfig(ctx, ref, opts)
+func (s *Solver) NewSecret(plaintext string) *dagger.Secret {
+	env := hashID(plaintext)
+	err := os.Setenv(env, plaintext)
```
Comment (jlongtine) [inline solver/solver.go:17]: Clean this up

**Comment by dolanor** [comment]
So it seems this PR is depending on some other work?
Is it possible to make it explicit in the OP? And maybe let it as a WIP until everything is available and implemented? Or use some label to document that? (so we can filter on the top level item that need to be merged first?)
